### ISSUE TITLE
Out-of-dictionary results returned

### ISSUE THREAD

**Issue by matanster** [universal-automata/liblevenshtein-java#10]
I have locally confirmed getting `findAll` results that are not in the dictionary provided, using the `Algorithm.TRANSPOSITION` algorithm. 

I think I could add an assertion for this case in [TranspositionTransducerTest.java](https://github.com/universal-automata/liblevenshtein-java/blob/41100a427414ecbb169b2495ff15ffb28b94ca16/src/test/java/com/github/dylon/liblevenshtein/levenshtein/TranspositionTransducerTest.java#L86-L100), where currently it seems that this condition is not being asserted against. Or is this already tested for in some other part of the test suite? Then try to reproduce on the existing test dictionary... 

For now unable to get existing tests working though..

**Comment by matanster** [comment]
Seems to equally apply to the `Algorithm.STANDARD`. 
Additional tidbit: in so far testing only a single out-of-dictionary find is included in each result of `findAll`.

**Comment by dylon** [comment]
What do you mean by, `findAll` results?  Also, please provide a test case that demonstrates the issue.

**Comment by matanster** [comment]
Sorry about that, it was the name of my Scala wrapper around `tranduce`. It will take a huge amount of creativity being able to demonstrate this without providing the proprietary lexicon... but it should be demonstrated in a reproducible way, I know. Lets put it on hold until I get to that.

**Comment by dylon** [comment]
Okay.  In the meantime, please test your wrapper against version 2.2.0, as the issue #6 was a pretty bad bug and may be the root cause of this.

**Comment by matanster** [comment]
On first examination, looks like the bug I reported here consistently reproduces with 2.1.2 but _not_ with 2.2.0. I will check again tomorrow. Any idea how the fix for #6 may have solved this?

**Comment by dylon** [comment]
Yes, I do.  Depending on the terms provided, some of the non-final nodes were overwriting the final ones, and vice-versa.  The way the final nodes were stored missed a few edge cases.  The tiny list I was using for testing happened to not have such terms, which is how the bug made it to production.

While debugging #6, I encountered similar behavior to what this issue describes, which is why I recognized it as the likely root cause.
